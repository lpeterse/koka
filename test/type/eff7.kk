fun while(p,f) {
  if (p()) then {
    f();
    while(p,f);
  }
  else {
    print("done");
    ()
  }
}