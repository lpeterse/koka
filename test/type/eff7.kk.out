while     : forall<a,e> (() -> <div,console|e> bool, () -> <div,console|e> a) -> <div,console|e> ()
